compiler/rustc_const_eval/src/interpret/machine.rs

```diff
@@ -681,14 +681,10 @@ pub macro compile_time_machine(<$tcx: lifetime>) {
     }
 
     #[inline(always)]
-    fn ub_checks(_ecx: &InterpCx<$tcx, Self>) -> InterpResult<$tcx, bool> {
-        // We can't look at `tcx.sess` here as that can differ across crates, which can lead to
-        // unsound differences in evaluating the same constant at different instantiation sites.
-        interp_ok(true)
-    }
-
-    #[inline(always)]
-    fn contract_checks(_ecx: &InterpCx<$tcx, Self>) -> InterpResult<$tcx, bool> {
+    fn runtime_checks(
+        _ecx: &InterpCx<$tcx, Self>,
+        _r: mir::RuntimeChecks,
+    ) -> InterpResult<$tcx, bool> {
         // We can't look at `tcx.sess` here as that can differ across crates, which can lead to
         // unsound differences in evaluating the same constant at different instantiation sites.
         interp_ok(true)
```

Comment:
this means that overflow checks using the intrinsic will always be run by ctfe, even when debug assertions and overflow checks are disabled. can you add a test for this?